misc/json-to-record-converter/README.md

```diff
@@ -0,0 +1,530 @@
+
+#### Scenario 12: JSON array with different JSON objects
+In this case even if the objects in the array are completely different(the fields), still those objects will be treated the same with differencing fields having optional types (In such scenario all fields will become optional) See _Scenario 13_.
+
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, {"name": "jerry", "age": 4, "country": "SL"}]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age;
+   string country?;
+};
+ 
+type NewRecord record {
+   ArrayPropertyItem arrayProperty;
+};
+```
+
+#### Scenario 13: JSON array completely different JSON objects
+This is where all the fields are different in the objects in an array. (Even though the fields are different all these objects will be treated as same when converting to Ballerina record)
+
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, {"city": "Kandy", "zip": 71500}]
+}
+```
+Ballerina
+```ballerina
+type ArrayProperty record {
+   string name?;
+   int age?;
+   string city?;
+   int zip?;
+};
+ 
+type NewRecord record {
+   ArrayProperty[] arrayProperty;
+};
+```
+
+#### Scenario 14: JSON array with different types
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, "String Value", 5, 5.67]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age?;
+};
+ 
+type NewRecordList record {
+   (ArrayPropertyItem|decimal|int|string)[] arrayProperty;
+};
+```
+In this case, if there are more than two types of values in an array. This will be handles as union of data types in Ballerina record creation.
+
+#### Scenario 15: JSON array of arrays
+This is where Nested JSON arrays are being handled.
+
+JSON
+```json
+{
+  "arrayProperty": [[10, 20, 40], [10, 20, 18]]
+}
+```
+Ballerina
+```ballerina
+type NewRecordList record {
+   int[][] arrayProperty;
+};
+```
+Here, the nested array can be any type, if the types are different, it would be handled differently. See _Scenario 17_
+
+#### Scenario 16: JSON array with different types and arrays
+JSON
+```json
+{
+  "arrayProperty": [{"name": "tom", "age": 5}, "String Value", 5, 5.67, [10, 20, 40], [["10", 20, 18]]]
+}
+```
+Ballerina
+```ballerina
+type ArrayPropertyItem record {
+   string name;
+   int age?;
+};
+ 
+type NewRecordList record {
+   (ArrayPropertyItem|decimal|int|string|int[]|(int|string)[][])[] arrayProperty;
+};
+```
+The implementation can handle such complex scenarios as well. Here there are many values with different data types are handled. There is a nested array of different types within the array. Nested array of the same type. JSON object and JSON primitives. And all these types are handled when creating a Ballerina record by having the union of all these types.
```

Comment:
```suggestion
The implementation can handle complex scenarios having many values with different data types as well. The above scenario has a nested array of different types within the array. The nested array is of the same type as the JSON object and JSON primitives. All these types are handled when creating a Ballerina record by having the union of all these types.
```
Please check the incomplete phrases and the suggestions to verify if the technical meaning has got changed.